src/context/Confirmation.tsx

```diff
@@ -0,0 +1,111 @@
+import {
+    Button,
+    Dialog,
+    DialogActions,
+    DialogContent,
+    DialogContentText,
+    DialogTitle,
+} from '@mui/material';
+import React, { createContext, useContext, useRef, useState } from 'react';
+import { FormattedMessage } from 'react-intl';
+
+export interface IConfirmationModalOptions {
```

Comment:
I am not a fan of the _I_ prefix on interfaces. They register as superfluous when an adopted pattern already distinguishes interfaces and types. Could you explain the value you find in this pattern?

Reply:
They make a lot of sense in context files like Confirm and Auth since they have a lot of stuff packed into 'em. That's why I used this here and no where else.